include/c_utilities/types/string_array.h

```diff
@@ -31,14 +31,38 @@ typedef struct C_UTILITIES_PUBLIC_TYPE string_array_t
   char ** data;
 } string_array_t;
 
+/// Returns an empty string array struct
+/*
+ * This function returns an empty and zero initialized string array struct.
+ * Calling utilities_string_array_fini on this instance leads to undefined
+ * behavior.
```

Comment:
As I understood it, this should not be the case. Instead this is what I expected:

```c
string_array_t foo;
utilities_string_array_fini(foo);  // undefined behavior

string_array_t bar = utilities_get_zero_initialized_string_array();
utilities_string_array_fini(bar);  // ok

string_array_t baz = utilities_get_pre_initialized_string_array(10);
utilities_string_array_fini(baz);  // also, ok
```